src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -244,6 +246,38 @@ public Task CleanTestHostAsync(CancellationToken cancellationToken)
             return Task.FromResult(true);
         }
 
+        private IEnumerable<string> FilterExtensionsBasedOnVersion(IEnumerable<string> extensions)
+        {
+            Dictionary<string, Tuple<Version, string>> extensionMap = new Dictionary<string, Tuple<Version, string>>();
+
+            foreach (var extensionFullPath in extensions)
```

Comment:
>foreach (var extensionFullPath in extensions) [](start = 12, length = 45)

Please try group by, and then get the highest file version within the group if the group size > 1